src/textual/message_pump.py

```diff
@@ -392,8 +393,16 @@ def post_message_from_child_no_wait(self, message: Message) -> bool:
             return False
         return self.post_message_no_wait(message)
 
-    async def on_callback(self, event: events.Callback) -> None:
-        await invoke(event.callback)
+    def on_callback(self, event: events.Callback) -> None:
+        self._callbacks.append(event.callback)
+
+    async def invoke_and_clear_callbacks(self) -> None:
+        """Invoke all callbacks that are waiting to be executed, and then clear them.
+        Callbacks will be invoked in the same order they were registered.
+        """
+        for callback in self._callbacks:
```

Comment:
Suspect it is entirely possible for this list to be modified during iteration. Suggest making a copy and then clearing it on self before invoking the callbacks.

Reply:
Good point